var/spack/repos/builtin/packages/python/package.py

```diff
@@ -919,11 +919,14 @@ def setup_dependent_build_environment(self, env, dependent_spec):
             env.prepend_path('PATH', path)
 
         python_paths = []
-        for d in dependent_spec.traverse(
-                deptype=('build', 'run', 'test')):
+        for d in dependent_spec.traverse(deptype=('build', 'run', 'test')):
             if d.package.extends(self.spec):
-                python_paths.append(join_path(d.prefix,
-                                              self.site_packages_dir))
+                # Python libraries may be installed in lib or lib64
+                # See issues #18520 and #17126
+                for lib in ['lib', 'lib64']:
+                    python_paths.append(join_path(
+                        d.prefix, lib, 'python' + str(self.version.up_to(2)),
+                        'site-packages'))
```

Comment:
Looking at the implementation, I think we can extract a simple method to compute the site packages directory:
```python
def site_packages(self, prefix):
    relative_path = os.path.join('python' + str(self.version.up_to(2)), 'site-packages')
    return os.path.join(prefix, relative_path)
```
and append only paths that exist to avoid cluttering:
```python
for lib in ('lib', lib64'):
    lib_prefix = os.path.join(d.prefix, lib)
    site_packages_dir = self.site_packages(lib_prefix)
    if not os.path.exists(site_packages_dir):
        continue
    python_paths.append(site_packages_dir)
````
Likewise below.

Reply:
@alalazo I think that's about the same number of lines of code either way.

The problem with using `os.path.exists` is that the path won't always exist but we still want it to be set. For example, when we are installing a new package, at the beginning of the build the directory will not exist, but the `PYTHONPATH` needs to be set in order to test the installation.

Reply:
> I think that's about the same number of lines of code either way.

Might be true, but the computation is done in a single place instead of two. Unless we adopt Martin Fowler's rule of three - in which case we should just wait to need one more duplication.

> The problem with using os.path.exists is that the path won't always exist but we still want it to be set. For example, when we are installing a new package, at the beginning of the build the directory will not exist, but the PYTHONPATH needs to be set in order to test the installation.

Hmm, can't we special case the root of the current DAG only:
```python
if package_is_root:
    # Add both lib and lib64 since it's still to be built
else:
    # Add only what exists
```
?

Reply:
Another alternative is to set this in the default modules.yaml prefix inspection.

Reply:
@alalazo do you have any thoughts on moving this to `modules.yaml`? Does `modules.yaml` support wildcards like `lib*/python*/site-packages`?

Reply:
I have to double check, but I don't think modules.yaml supports globbing. I'll look into what it takes to add that support.

Reply:
I guess that would only address the issue for `setup_run_environment`, not `setup_dependent_build_environment`.